ArduSub/Sub.h

```diff
@@ -335,17 +336,17 @@ class Sub : public AP_Vehicle {
     // Inertial Navigation
     AP_InertialNav_NavEKF inertial_nav;
 
-    AP_AHRS_View ahrs_view;
+    AP_AHRS_View *ahrs_view;
 
     // Attitude, Position and Waypoint navigation objects
     // To-Do: move inertial nav up or other navigation variables down here
-    AC_AttitudeControl_Sub attitude_control;
+    AC_AttitudeControl_Sub *attitude_control;
 
-    AC_PosControl_Sub pos_control;
+    AC_PosControl_Sub *pos_control;
 
-    AC_WPNav wp_nav;
-    AC_Loiter loiter_nav;
-    AC_Circle circle_nav;
+    AC_WPNav *wp_nav;
+    AC_Loiter *loiter_nav;
+    AC_Circle *circle_nav;
```

Comment:
you should not have to change all of these.

Reply:
When I revert this I get the following error: 
![image](https://user-images.githubusercontent.com/20564040/200278908-9fa0eb0b-5642-4faa-afc9-5311f0831279.png)

I thought this might be related to something not being allocated, that was the reason I changed all of these.

Reply:
You need to convert to a pointer here:
https://github.com/ArduPilot/ardupilot/blob/1488edb1d807888dd6166106d8888fd236d810a4/ArduSub/Sub.cpp#L35-L36

eg:
```
          attitude_control(ahrs_view, aparm, *motors, scheduler.get_loop_period_s()),
          pos_control(ahrs_view, inertial_nav, *motors, attitude_control, scheduler.get_loop_period_s()),
```
Although there may still be a ordering issue.

Reply:
What do you mean by "ordering issue"? 
The order the individuals of each class are initialized?

Reply:
Motors will be nullptr in the at construction. `attitude_control` probably does not check. So if it trys to use before we have constructed motors it will fail.  Copter construct's motors first and will not progress if that fails.

Reply:
So, in this case, in [AC_AttitudeControl_Sub](https://github.com/ArduPilot/ardupilot/blob/74f8ceeb418d8e0becdcb7a43e0e2b2e2d65b868/libraries/AC_AttitudeControl/AC_AttitudeControl_Sub.h#L74) `_motors_multi` is initialized with the address of the nullptr, and after motors is initialized this address does not change. So, `attitude_control` must be initialized later? And I guess the other variables as well?

Reply:
I added some checks to verify if the vehicle is initialized (copying from Copter) but I still get the error in the first post of this PR. I also tried with reverting the variables mentioned in this thread to non-pointers, but I still get the error mentioned in this thread.